# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/loader/URLPreloader.h
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 js/xpconnect/loader/URLPreloader.h
--- a/js/xpconnect/loader/URLPreloader.h	Mon Sep 14 17:32:40 2020 +0000
+++ b/js/xpconnect/loader/URLPreloader.h	Mon Sep 14 17:00:53 2020 +0000
@@ -26,6 +26,8 @@
 #include "nsIThread.h"
 #include "nsReadableUtils.h"
 
+class nsZipArchive;
+
 namespace mozilla {
 namespace loader {
 class InputBuffer;
@@ -75,7 +77,7 @@
   static Result<nsCString, nsresult> ReadFile(nsIFile* file,
                                               ReadType readType = Forget);
 
-  static Result<nsCString, nsresult> ReadZip(CacheAwareZipReader* archive,
+  static Result<nsCString, nsresult> ReadZip(nsZipArchive* archive,
                                              const nsACString& path,
                                              ReadType readType = Forget);
 
@@ -195,7 +197,7 @@
       return "";
     }
 
-    already_AddRefed<CacheAwareZipReader> Archive() {
+    already_AddRefed<nsZipArchive> Archive() {
       return Omnijar::GetReader(OmnijarType());
     }
 